libs/feature/router/src/lib/default/services/router-search.service.spec.ts

```diff
@@ -1,19 +1,24 @@
 import { BehaviorSubject } from 'rxjs'
 import { RouterSearchService } from './router-search.service'
 
-const state = { Org: { mel: true } }
-const searchFacadeMock: any = {
-  searchFilters$: new BehaviorSubject(state),
+let state = {}
+class SearchFacadeMock {
+  searchFilters$ = new BehaviorSubject(state)
 }
-const routerFacadeMock: any = {
-  setSearch: jest.fn(),
-  updateSearch: jest.fn(),
+class RouterFacadeMock {
+  setSearch = jest.fn()
+  updateSearch = jest.fn()
 }
 describe('RouterSearchService', () => {
   let service: RouterSearchService
+  let routerFacade: RouterFacadeMock
+  let searchFacade: SearchFacadeMock
 
   beforeEach(() => {
-    service = new RouterSearchService(searchFacadeMock, routerFacadeMock)
+    state = { OrgForResource: { mel: true } }
+    routerFacade = new RouterFacadeMock()
+    searchFacade = new SearchFacadeMock()
+    service = new RouterSearchService(searchFacade as any, routerFacade as any)
```

Comment:
For my understanding, why was it necessary to change the mocks to classes here?

Reply:
I tend to change value mocks to class mocks because value mocks can introduce side-effects between tests, e.g. running tests in a different order might yield different results (since the same object is used multiple times and potentially modified)

Reply:
ok, thanks for the explanation :-)